userland/libtock/ble.h

```diff
@@ -2,125 +2,120 @@
+  SIMULTANEOUS_LE_BREDR_H = 0x10 /*Not relevant - central mode only. */
+} GapFlags_t;
+
+
+
+
+/*******************************************************************************
+ *   User-space API
+ ******************************************************************************/
+
+
+ // initialize advertisement (should be used before invoken ble_start_advertising)
+ //
+ // advertising_itv_ms  - advertisement interval in milliseconds
+ // flags               - flags in the advertisement defined in GapFlags_t enum
+ // addr                - address to be used in the advertisment
+ // addr_len            - length of the address (it should be 6)
+int ble_initialize(uint16_t advertising_itv_ms, uint8_t flags, uint8_t *addr, uint8_t addr_len);
+
+// start advertising
+int ble_start_advertising(void);
+
+
+// stop advertising but don't change anything in the packet configuration
+int ble_stop_advertising(void);
+
+// reset the entire advertisement data packet
+// excluding flags and advertisement address
+int ble_reset_advertisement(void);
+
+
+ // configure advertisement name
+ //
+ // device_name         - device named to be used in the advertisement
+ // len                 - size in bytes
+int ble_advertise_name(uint8_t *device_name, uint8_t size_b);
+
+// configure list of 16 bit uuids
+//
+// uuid16               - array of 16 bit uuids
+// size                 - size in bytes
+int ble_advertise_uuid16(uint16_t *uuid16, uint8_t size_b);
 
-#ifdef __cplusplus
-extern "C" {
-#endif
 
-// configure advertisement data
+// configure service data
 //
-// type     - ad type
-// data     - buffer with data
-// len      - length of the data buffer
-int ble_adv_data(uint8_t type, const unsigned char *data, uint8_t len);
+// uuid16               - 16 bit uuid to be associated with the data
+// data                 - array of data in bytes
+// data_len             - size in bytes
+int ble_advertise_service_data(uint16_t uuid16, uint8_t *data, uint8_t size_b);
 
-// clear the advertisement data
-int ble_adv_clear_data(void);
 
-// configure tx power
+// configure manufacturer specific data
 //
-// power    - tx power in DBm (-30 to +4)
-int ble_adv_set_txpower(BLE_TX_Power_t power);
+// data                 - array of data in bytes
+// data_len             - size in bytes
+int ble_advertise_manufacturer_specific_data(uint8_t *data, uint8_t size_b);
 
 
-// configure advertisement interval
+// passive scanning of advertisements
 //
-// interval   - advertisement interval in milliseconds (20ms - 10240ms)
-int ble_adv_set_interval(uint16_t interval);
-
-// start send BLE advertisement periodically according to the configured interval
-// if no interval is connected it will use a pre-configured value
+// data                 - array of bytes to write the received advertisment to
+// len                  - size in bytes of data
+// callback             - callback handler to call when an advertisement is received
```

Comment:
What's the type signature of the callback handler?

Reply:
```c
static void callback(__attribute__((unused)) int unused0,
                     __attribute__((unused)) int unused1, __attribute__((unused)) int unused2,
__attribute__((unused)) void* ud);
```
It was like that before at least

Reply:
I'd suggest documenting in that header file what the parameters in the callback actually mean. If I was going to write code that scanned for advertisements, I wouldn't know what the parameters I get in the callback actually mean without reading through the kernel code.